pandas/core/indexes/api.py

```diff
@@ -206,6 +206,17 @@ def conv(i):
 
         return Index(lib.fast_unique_multiple_list([conv(i) for i in inds], sort=sort))
 
+    # GH 35092. Detect if we have an Index type, for which the sort
+    # setting doesn't make sense
+    ind_types = list({type(index) for index in indexes})
+    if any(
+        ind_type in [MultiIndex, RangeIndex, DatetimeIndex, CategoricalIndex]
+        for ind_type in ind_types
+    ):
+        ignore_sort = True
+    else:
+        ignore_sort = False
```

Comment:
If we let any of these index type not get sorted, we'll break some tests where we assume that the results of joining on any of these come out sorted. Moreover, there isn't much point to having any of them unsorted except for pretty printing purposes, in my opinion.
If you think we should pass sort through for every type of `Index` subclass, please let me know.

Reply:
can you show what tests break?

Reply:
i would rather just pass thru and adjust the tests

Reply:
done